lib/main.js

```diff
@@ -18,14 +19,22 @@ Client.prototype.apiCall = function (method, path, params, callback) {
     'Host': this.host,
     'User-Agent': `duo_api_nodejs/${_PACKAGE_VERSION}`
   }
-  headers['Authorization'] = duo_sig.sign(
-    this.ikey, this.skey, method, this.host, path, params, date)
-
-  var qs = querystring.stringify(params)
   var body = ''
+  var qs = querystring.stringify(params)
+  if (this.sigVersion === 5) {
+    if (method === 'POST' || method === 'PUT') {
+      body = JSON.stringify(params)
+      params = {}
+    }
+    headers['Authorization'] = duo_sig.signV5(
+      this.ikey, this.skey, method, this.host, path, params, date, body)
+  } else {
+    headers['Authorization'] = duo_sig.sign(
+      this.ikey, this.skey, method, this.host, path, params, date)
+  }
+
   if (method === 'POST' || method === 'PUT') {
-    body = qs
-    headers['Content-type'] = 'application/x-www-form-urlencoded'
+    headers['Content-type'] = 'application/json'
```

Comment:
I believe the content type for a v2 with a body would still be `application/x-www-form-encoded`?

that's my reading of https://github.com/duosecurity/duo_client_python/blob/master/duo_client/client.py#L350 at least...

Reply:
and I think the params need to be encoded and put in the body here still in the v2 case?    I don't see anything that sets the body in v2 + PUT/POST?